.github/workflows/test.yml

```diff
@@ -42,6 +42,11 @@ jobs:
         shell: bash
       - name: Build workspace
         run: catkin build --summarize --no-status
+      - name: Build CHIMP
```

Comment:
Should building CHIMP be a part of building ROSPlan? No other planners are compiled as a part of compiling ROSPlan - indeed, it would be too much to ask a user of ROSPlan to resolve all dependencies of and compile all planners, even though they would normally only be interested in using one.

Reply:
Why isn't it built as part of rosplan dependencies as it's defined as one? Rddlsim for instance is built that way during catkin_make.

Reply:
I am actually building it here is so that it does not have to be compiled by all users:
Because CHIMP uses gradle it normally does not need to be compiled in advance.
However, the first time you call CHIMP via gradle it is significantly slower (especially with the github action) such that the tests failed due to a timeout. Therefore, for the CI I am building it in advance such that gradle downloads dependencies and compiles chimp - making the subsequent tests pass.
A normal user does not have to build CHIMP - especially not if he does not want to use it.

If you prefer to build it during catkin_make I can try to put it there.

Reply:
Ok, I see this is only for the test. So it is not being built by default?
I understand CHIMP only needs to build only once, while the workspace is building. There should not be a timeout there during the catkin build. I'm not familiar with gradle so I am not sure about. 
Maybe we could add a CMake flag to compile CHIMP or not, so we can specify whether to build or not during catkin build? And then we set it for the GitHub actions. This may be easier for users than having to build chimp manually and compiling two things.

Reply:
The timeout does not happen in the catkin build, but in the first test of the planner interface for CHIMP. It is part of the planner command that the planner inferface uses to call CHIMP: `timeout 50 " + chimp_path + "/gradlew run -p " + chimp_path + " -Dexec.args=\"DOMAIN PROBLEM -o=OUTPUT\"`
This only fails on the slow CI server.

Your suggestion to build it dependent on a flag with catkin sounds good and it might be the cleaner solution for this.

Reply:
With 2001212 CHIMP is built by catkin if the cmake variable `ROSPLAN_COMPILE_CHIMP` is set to `TRUE` as suggested by @gerardcanal.

Reply:
This looks great Sebastian! I think we should probably do the same with RDDLSim and the other things that compile there. I'll open an issue to remember this.